processor/samplingprocessor/processor.go

```diff
@@ -0,0 +1,64 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package samplingprocessor
+
+import (
+	"context"
+	"math/rand"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type throughputMeasurementProcessor struct {
+	logger          *zap.Logger
+	dropCutOffRatio float64
+}
+
+func newThroughputMeasurementProcessor(logger *zap.Logger, cfg *Config) *throughputMeasurementProcessor {
+	return &throughputMeasurementProcessor{
+		logger:          logger,
+		dropCutOffRatio: cfg.DropRatio,
+	}
+}
+
+func (tmp *throughputMeasurementProcessor) processTraces(_ context.Context, td ptrace.Traces) (ptrace.Traces, error) {
+	//#nosec G404 -- randomly generated number is not used for security purposes. It's ok if it's weak
+	if rand.Float64() <= tmp.dropCutOffRatio {
+		return ptrace.NewTraces(), nil
+	}
+
+	return td, nil
+}
+
+func (tmp *throughputMeasurementProcessor) processLogs(_ context.Context, ld plog.Logs) (plog.Logs, error) {
+	//#nosec G404 -- randomly generated number is not used for security purposes. It's ok if it's weak
+	if rand.Float64() <= tmp.dropCutOffRatio {
+		return plog.NewLogs(), nil
+	}
+
+	return ld, nil
+}
```

Comment:
Should we be doing this for each individual log record, instead of the whole plog.Logs? I think this would end up with the distribution being less uniform over shorter periods of time. Maybe that isn't a concern, though.

The metrics and traces processors make sense to drop whole payloads, I think (metrics could maybe drop per-ResourceMetric, but it makes sense to just eat the whole output of the scraper).

Reply:
Yeah good call. I'll make that change.